AutomaticQC/SpikeClassifiers/imosSpikeClassifiersLimits.m

```diff
@@ -0,0 +1,119 @@
+% This program is free software: you can redistribute it and/or modify
+% it under the terms of the GNU General Public License as published by
+% the Free Software Foundation version 3 of the License.
+%
+% This program is distributed in the hope that it will be useful,
+% but WITHOUT ANY WARRANTY; without even the implied warranty of
+% MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+% GNU General Public License for more details.
+%
+% You should have received a copy of the GNU General Public License
+% along with this program.
+% If not, see <https://www.gnu.org/licenses/gpl-3.0.en.html>.
+%
+
+limits = struct();
+limits.hampel_half_window_width.min = int64(1);
+limits.hampel_half_window_width.max = int64(len / 2);
+limits.hampel_half_window_width.help = '[Int] such as <half_window_width,1,half_window_width>. Increase this value to include more data in the window';
+
+limits.hampel_madfactor.min = .5;
+limits.hampel_madfactor.max = 20;
+limits.hampel_madfactor.help = '[float] A scale multipler for the MAD <N*mad>. Increase this value to reduce sensitivity';
+
+limits.hampel_lower_mad_limit.min = 0.;
+limits.hampel_lower_mad_limit.max = 0.5;
+limits.hampel_lower_mad_limit.help = '[float] Clip MAD values to above this limit. Increase slightly to reduce sensitivity for low/constant regions';
+
+limits.otsu_savgol_window.min = int64(3);
+limits.otsu_savgol_window.max = int64(len / 3);
+limits.otsu_savgol_window.help = '[int] The full window width for the SavGol low-pass filter. Increase the window to increase the number of noise considered. Too large values affect resolution at the signal boundaries';
+
+limits.otsu_savgol_pdeg.min = int64(1);
+limits.otsu_savgol_pdeg.max = int64(4);
+limits.otsu_savgol_pdeg.help = '[int] The least squares polynomial order. A larger value may increase noise level but move spike locations.';
+
+limits.otsu_savgol_nbins.min = int64(2);
+limits.otsu_savgol_nbins.max = int64(256);
+limits.otsu_savgol_nbins.help = '[int] The number of histogram bins when tracing the noise level. A larger value increase spike magnitude resolution at the window level.';
+
+limits.otsu_savgol_oscale.min = 0.01;
+limits.otsu_savgol_oscale.max = 10;
+limits.otsu_savgol_oscale.help = '[float] A scale factor to multiply the noise threshold computed on the noise level. A large value increase the threshold (reduce detection).';
+
+limits.otsu_threshold_nbins.min = int64(2);
+limits.otsu_threshold_nbins.max = int64(256);
+limits.otsu_threshold_nbins.help = '[int] The number of histogram bins when tracing the noise level. A larger value reduce global spike magnitude resolution.';
+
+limits.otsu_threshold_oscale.min = 1e-5;
+limits.otsu_threshold_oscale.max = 10;
+limits.otsu_threshold_oscale.help = '[float] A scale factor to multiply the noise threshold computed on the noise level. A large value increase the threshold (reduce detection).';
+
+limits.otsu_threshold_centralise.min = false;
+limits.otsu_threshold_centralise.max = true;
```

Comment:
why are these booleans in this part? is it just to have 0 and 1 ? why not int64(false) ?

Reply:
I handle specific types!That's why